### PR TITLE
ref(event-entries): Improved Event Entries type render

### PR DESCRIPTION
**Pull request by priscilawebdev**
Introduced the EventEntry component, which uses a switch method to render each interface with the correct props. Before, the UI was passing all the props for all the interfaces, even if some props were not needed.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/events/eventEntry.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  event: Event;
+  organization: SharedViewOrganization | Organization;
+  isShare: boolean;
+};
+
+function EventEntry({entry, event, projectSlug, organization}: Props) {
+  switch (entry.type) {
+    case EntryType.EXCEPTION: {
+      const {data, type} = entry;
+      return <Exception type={type} event={event} data={data} projectId={projectSlug} />;
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/events/eventEntry.tsx:29]: Would it be easier to destructure the entire `entity` above and write these as 

```tsx
case EntryType.EXCEPTION: {
	return <Exception {...{type, event, data, projectId}} />;
}
case EntryType.MESSAGE: {
  	return <Message {...{data}} />;
}
	
case EntryType.REQUEST: {
	  return <Request {...{type, event, data}} />;
}
```

just my thoughts, but I think that is easier to read.

Comment (priscilawebdev) [reply src/sentry/static/sentry/app/components/events/eventEntry.tsx:29]: I prefer to have it separated 🤷‍♀️ but I don't have strong feelings .. @getsentry/app-frontend please what do you think about it?

Comment (billyvg) [reply src/sentry/static/sentry/app/components/events/eventEntry.tsx:29]: The props vary so much that I would prefer them separated as well, but I don't have strong feelings either way.

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/components/events/eventEntry.tsx:29]: Cool, let's leave it 👍

**Review by dashed — APPROVED** [review]

**Pull request merged**